### DISCUSSION TITLE
Type errors when following Authentication and Access guide from the docs.

### DISCUSSION THREAD

**Discussion by ghost** [keystonejs/keystone#9336] [Questions] [answered]
Hello!

I'm following the [guide for Authentication and Access](https://keystonejs.com/docs/guides/auth-and-access-control) and I'm getting the following type error:
`Type '({ session }: { session: Session; }) => boolean' is not assignable to type 'ListOperationAccessControl<"query", BaseListTypeInfo>'.
  Types of parameters '__0' and 'args' are incompatible.
    Type 'BaseAccessArgs<BaseListTypeInfo> & { operation: "query"; }' is not assignable to type '{ session: Session; }'.
      Property 'session' is optional in type 'BaseAccessArgs<BaseListTypeInfo> & { operation: "query"; }' but required in type '{ session: Session; }'.`

It comes from this code from the guide:

```
const isAdmin = ({ session }: { session: Session }) => Boolean(session?.data.isAdmin);

const Post = list({
  access: {
    operation: {
      query: isAdmin,
      create: isAdmin,
      update: isAdmin,
      delete: isAdmin,
    },
  },
  fields: {
    // see above
  },
});
```

The project still runs. I'm new to TypeScript, so I'm not sure if this is something I can fix with a setting or if this is a documentation issue that needs to be addressed.

**Comment by dcousens** [comment, marked as answer]
> Property 'session' is optional in type 'BaseAccessArgs<BaseListTypeInfo> & { operation: "query"; }' but required in type '{ session: Session; }'.

The `session: Session` type should be optional `session?: Session`.
The documentation is in error here, and has been fixed in https://github.com/keystonejs/keystone/commit/bf300448b6b15914c7fe70ddc0b147ceaf9049ff :heavy_check_mark:.

Thanks for questioning this @indexofmetals! :yellow_heart:
